src/test/scala/ai/privado/languageEngine/kotlin/tagger/collection/KotlinCollectionTaggerTest.scala

```diff
@@ -0,0 +1,212 @@
+          |        }
+          |    }.start(wait = true)
+          |}
+          |""".stripMargin
+        cpg = buildCpg(TestCodeSnippet(sourceCode = fileContents, language = Language.KOTLIN))
+        cpg.call.methodFullName(".*.get.*").l.size shouldBe 1
+        cpg.call.head.code.contains("get") shouldBe true
+
+        val collectionTagger = new MethodFullNameCollectionTagger(cpg, ruleCacheWithCollectionRule(collectionRule))
+        collectionTagger.createAndApply()
+
+        val ingressRules = collectionTagger.getIngressUrls()
+        ingressRules should contain("\"/hello\"")
+
+        val callNode = cpg.call.methodFullName(".*.get.*").head
+        callNode.name shouldBe "get"
+        val tags = callNode.argument.isMethodRef.head.referencedMethod.tag.l
+        tags.size shouldBe 6
+        tags.nameExact(Constants.id).head.value shouldBe ("Collections.Kotlin.HttpFramework")
+        tags.nameExact(Constants.catLevelOne).head.value shouldBe Constants.collections
+        tags.nameExact(Constants.catLevelTwo).head.value shouldBe Constants.default
+        tags.nameExact(Constants.nodeType).head.value shouldBe "REGULAR"
+        tags.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/hello\""
+      } finally {
+        if (cpg != null) {
+          cpg.close()
+        }
+      }
+    }
+    "tag get Spark Java collection endpoint with method handler as parameter" in {
+      var cpg: Cpg = null
+      try {
+        val fileContents =
+          """
+          |import spark.Spark
+          |
+          |fun main(args: Array<String>) {
+          |    Spark.get("/hello") { req, res ->
+          |        "Hello Spark Kotlin!"
+          |    }
+          |}
+          |""".stripMargin
+        cpg = buildCpg(TestCodeSnippet(sourceCode = fileContents, language = Language.KOTLIN))
+
+        val ruleCache        = ruleCacheWithCollectionRule(collectionRule)
+        val collectionTagger = new MethodFullNameCollectionTagger(cpg, ruleCache)
+        collectionTagger.createAndApply()
+
+        val ingressRules = collectionTagger.getIngressUrls()
+        ingressRules should contain("\"/hello\"")
+
+        val getCalls = cpg.call.methodFullName(".*get.*").l
+        getCalls should have size 1
+        getCalls.name.toSeq should contain theSameElementsAs List("get")
+
+        val tags = getCalls.head.argument.isMethodRef.head.referencedMethod.tag.l
+        tags should have size 6
+        tags.nameExact(Constants.id).head.value shouldBe ("Collections.Kotlin.HttpFramework")
+        tags.nameExact(Constants.catLevelOne).head.value shouldBe Constants.collections
+        tags.nameExact(Constants.catLevelTwo).head.value shouldBe Constants.default
+        tags.nameExact(Constants.nodeType).head.value shouldBe "REGULAR"
+        tags.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/hello\""
+
+        // assert collection exporter
+        val collectionExporter   = new CollectionExporter(cpg, ruleCache)
```

Comment:
Adding CollectionExporter test to only one of each test in Kotlin and Java since it becomes repetitive.